addons/udes_stock/models/stock_move_line.py

```diff
@@ -1110,14 +1115,35 @@ def _drop_off_criterion_summary(self):
         "qty_done",
     )
     def _prevent_write_after_done(self):
-        done_lines = self.filtered(lambda x: x.state == "done")
+        done_lines = self.filtered(lambda ml: ml.state == "done")
+
+        # When a moveline is updated through desktop only one "done" move line is passed into self even when there are multiple
+        # Get all the done move lines on the picking if there is only one "done" move line
+        if len(done_lines) == 1 and self.mapped("picking_id"):
+            done_lines = self.mapped("picking_id.move_line_ids").filtered(
```

Comment:
Suggestion: This mapped could be pretty expensive on big databases, and this constraint will be checked fairly often.

Can we try using `search` or `search_read` here?

Reply:
I had thought about this, but I didn't think there would be much of a performance dip as the mapped will only ever be called if self contains `done_lines` which might be quite rare.   

Happy to change it though